Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/CallBAP.hs

```diff
@@ -154,6 +158,10 @@ sendRideAssignedUpdateToBAP booking ride = do
       >>= fromMaybeM (MerchantNotFound booking.providerId.getId)
   driver <- QPerson.findById ride.driverId >>= fromMaybeM (PersonNotFound ride.driverId.getId)
   vehicle <- QVeh.findById ride.driverId >>= fromMaybeM (VehicleNotFound ride.driverId.getId)
+  imagePath <- ((.compAadhaarImagePath) =<<) <$> CQDI.findById (cast ride.driverId)
+  image <- case imagePath of
+    Nothing -> return Nothing
+    Just path -> Just <$> S3.get (unpack path)
```

Comment:
This part also you need to cache.
Otherwise you will keep on getting it from S3.
Get it once and cache the image.